### PR TITLE
all: Unused args, retvals, assignments

### PR DESCRIPTION
**Pull request by greatroar**
Also two typos. These were all found by golangci-lint or DeepSource.

### CODE REVIEW

File: lib/db/transactions.go [hunk 1]
```diff
@@ -714,7 +713,7 @@ func (t readWriteTransaction) updateGlobal(gk, keyBuf, folder, device []byte, fi
 			// The inserted file is the global file
 			global = file
 		} else {
-			keyBuf, global, err = t.getGlobalFromFileVersion(keyBuf, folder, name, true, globalFV)
+			keyBuf, global, _ = t.getGlobalFromFileVersion(keyBuf, folder, name, true, globalFV)
```
Comment (calmh) [inline lib/db/transactions.go:716]: I suspect that we should in fact have some sort of error handling here, so the bug is larger than just unused retval

Comment (imsodin) [reply lib/db/transactions.go:716]: Generally I want to have a good look at all this db changes, maybe there's some insights (bugs) to be discovered.

Comment (imsodin) [reply lib/db/transactions.go:716]: On a second, slower look it all looks entirely benign. Just as noted here where a `if err != nil { return ..., err}` has been forgotten previously.

File: cmd/syncthing/main.go [hunk 2]
```diff
@@ -516,7 +516,7 @@ func checkUpgrade() (upgrade.Release, error) {
 }
 
 func upgradeViaRest() error {
-	cfg, _ := loadOrDefaultConfig(protocol.EmptyDeviceID, events.NoopLogger, true)
+	cfg, _ := loadOrDefaultConfig(protocol.EmptyDeviceID, events.NoopLogger)
```
Comment (AudriusButkevicius) [inline cmd/syncthing/main.go:519]: probably should have an error check?

File: lib/db/transactions.go (hunk 1 above)
Comment (greatroar) [reply lib/db/transactions.go:716]: Inserted an error check.

File: cmd/syncthing/main.go (hunk 2 above)
Comment (greatroar) [reply cmd/syncthing/main.go]: Done.

**Review by imsodin — APPROVED** [review]

**Pull request merged**